x-pack/plugins/cloud_security_posture/public/pages/vulnerabilities/vulnerabilities.tsx

```diff
@@ -321,112 +303,146 @@ const VulnerabilitiesContent = ({ dataView }: { dataView: DataView }) => {
-              opacity: isFetching ? 1 : 0,
-            }}
-          />
-          <EuiDataGrid
-            className={cx({ [styles.gridStyle]: true }, { [styles.highlightStyle]: showHighlight })}
-            aria-label={VULNERABILITIES}
-            columns={columns}
-            columnVisibility={{
-              visibleColumns: columns.map(({ id }) => id),
-              setVisibleColumns: () => {},
-            }}
-            schemaDetectors={[severitySchemaConfig]}
-            rowCount={limitedTotalItemCount}
-            toolbarVisibility={{
-              showColumnSelector: false,
-              showDisplaySelector: false,
-              showKeyboardShortcuts: false,
-              showFullScreenSelector: false,
-              additionalControls: {
-                left: {
-                  prepend: (
-                    <>
-                      <EuiButtonEmpty size="xs" color="text">
-                        {i18n.translate('xpack.csp.vulnerabilities.totalVulnerabilities', {
-                          defaultMessage:
-                            '{total, plural, one {# Vulnerability} other {# Vulnerabilities}}',
-                          values: { total: data?.total },
-                        })}
-                      </EuiButtonEmpty>
-                    </>
-                  ),
-                },
-                right: (
-                  <EuiFlexItem grow={false} className={styles.groupBySelector}>
-                    <FindingsGroupBySelector
-                      type="default"
-                      pathnameHandler={vulnerabilitiesPathnameHandler}
-                    />
-                  </EuiFlexItem>
-                ),
-              },
-            }}
-            gridStyle={{
-              border: 'horizontal',
-              cellPadding: 'l',
-              stripes: false,
-              rowHover: 'none',
-              header: 'underline',
-            }}
-            renderCellValue={renderCellValue}
-            inMemory={{ level: 'enhancements' }}
-            sorting={{ columns: sort, onSort: onSortHandler }}
-            pagination={{
-              pageIndex,
-              pageSize,
-              pageSizeOptions: [10, 25, 100],
-              onChangeItemsPerPage,
-              onChangePage,
-            }}
-          />
-          {isLastLimitedPage && <LimitedResultsBar />}
-          {showVulnerabilityFlyout && selectedVulnerability && (
-            <VulnerabilityFindingFlyout
-              flyoutIndex={selectedVulnerabilityIndex}
-              vulnerabilityRecord={selectedVulnerability}
-              totalVulnerabilitiesCount={limitedTotalItemCount}
-              onPaginate={onPaginateFlyout}
-              closeFlyout={onCloseFlyout}
-              isLoading={isFetching}
-            />
-          )}
-        </>
+      {error && <ErrorCallout error={error as Error} />}
+      {!error && (
+        <VulnerabilitiesDataGrid dataView={dataView} data={data} isFetching={isFetching} />
```

Comment:
most of the code changes are moving the DataGrid into its own component to avoid complex inline logic on when to render and not to render. There is a lot of similarity between 3 pages in Vulnerabilities, we might want to refactor these in the future, but I think it's out of scope for this ticket. Let me know if you think otherwise